### PR TITLE
Time smoothing

### PR DESCRIPTION
**Pull request by Darel13712**
Three types of smoothing: power, exponential and linear

### CODE REVIEW

File: replay/time.py [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    |      2|2099-03-22 00:00:00|0.9333333333333333|
+    |      3|2099-03-26 00:00:00|               1.0|
+    +-------+-------------------+------------------+
+    <BLANKLINE>
+
+    This function **does not** take relevance values of interactions into account.
+    Only item age is used.
+
+    :param log: interactions log
+    :param decay: number of days after which the weight is reduced by half, must be grater than 1
```
Comment (AleXXL1986) [inline replay/time.py:77]: Why do you use this approach? I think usually users use age and coef

**Review by AleXXL1986 — CHANGES_REQUESTED** [review]

File: replay/time.py [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+def get_item_recency(
+    log: AnyDataFrame,
+    decay: float = 30,
+    limit: float = 0.1,
+    kind: str = "power",
+):
+    """
+    Calculate item weight showing when the majority of interactions with this item happened.
+
+    >>> import pandas as pd
```
[the remark this answers is not in the document]
Comment (Darel13712) [reply replay/time.py:17]: But this is not just random tests. this is a part of documentation

File: replay/time.py [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+
+from replay.utils import convert2spark
+from replay.constants import AnyDataFrame
+
+
+def get_item_recency(
+    log: AnyDataFrame,
+    decay: float = 30,
+    limit: float = 0.1,
+    kind: str = "power",
```
[the remark this answers is not in the document]
Comment (Darel13712) [reply replay/time.py:12]: but why?

File: replay/time.py (hunk 1 above)
Comment (Darel13712) [reply replay/time.py:77]: What approach?

File: replay/time.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AleXXL1986) [reply replay/time.py:17]: yes, but in start of docstring user want to see method description. For example look at pyspark code

File: replay/time.py (hunk 1 above)
Comment (AleXXL1986) [reply replay/time.py:77]: i mean use this decay. it is not intuitive in all cases

File: replay/time.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (AleXXL1986) [reply replay/time.py:12]: because it more logical, and similar on step change in NN

File: replay/time.py (hunk 1 above)
Comment (Darel13712) [reply replay/time.py:77]: Well it is intuitive. And more importantly it is universal for all the methods. do you suggest a user to provide three semantically different parameters under the same name? power or base are easy mathematically but are not intuitive, you won't know how fast it is decreasing, but with decay you know exact point where weight is 0.5

**Review by AleXXL1986 — APPROVED** [review]

**Pull request merged**